packages/web/src/pages/campaign.js

```diff
@@ -0,0 +1,114 @@
+// Module imports
+import { useCallback, useEffect} from 'react'
+import { useRouter } from 'next/router'
+
+// Local imports
+import { Field } from 'components/Field'
+import { Form } from 'components/Form'
+import { FormButton } from 'components/FormButton'
+import { useAuth } from 'contexts/AuthContext'
+import {firestore, firebase} from '../helpers/firebase'
+
+
+export default function CampaignPage() {
+
+	const {
+		isLoggedIn
+		,user
+	} = useAuth()
+
+	const router = useRouter()
+	useEffect(() => 
+	{
+		if (!isLoggedIn)
+		{
+			router.push('/login');
+		}
+	})
+
+
+	const handleSubmit = useCallback(formState => {
+		const {
+			isValid,
+			values,
+		} = formState
+
+		if (isValid) {
+			ValidateAndSaveCampaign(values, user, router)
+		}
+	}, [ValidateAndSaveCampaign, user])
+
+	return (
+		<div>
+			<header><h2>Create Campaign</h2></header>
+			<Form 
+				initialValues={{
+					name: '',
+					description: ''
+				}}
+			onSubmit={handleSubmit}>
+			<Field
+					id="name"
+					isRequired
+					label="Name"
+					type="text"
+					maxLength={50} />
+
+				<Field
+					id="description"
+					isRequired
+					label="Description"
+					type="text"
+					maxLength={300} />
+
+				<FormButton
+					type="submit"
+					isDisabled={!isLoggedIn}>
+					Create
+				</FormButton>
+			</Form>
+		</div>
+		
+	)
+}
+
+function ValidateAndSaveCampaign(campaign, user, router)
+{
+	if (user.uid && campaign.description && campaign.name)
+	{
+		let isFirst = false;
+    	var campaignsCollection = firestore.collection('campaigns');
+		campaignsCollection
+			.where("ownerID", "==", user.uid)
+			.get()
```

Comment:
I'm not sure I follow... What's the point of retrieving the user's campaigns before creating the new one?

Reply:
The original issue request stated that we should only set the isActive flag to true if the campaign created is the first campaign ever created by a user. So we check whether any campaigns exist before adding a campaign to determine the value for this flag.

Presumably, only 1 campaign should be active at a time so additional development is needed to switch the active campaign.

Reply:
Aaaaah, that makes sense. To prevent retrieving documents unnecessarily, you can add another `.where` for `isActive`. Currently, it'll attempt to retrieve all campaigns that belong to the user, potentially incurring unnecessary reads. Adding a filter for only active campaigns will ensure that we never get more than one result.

Reply:
Also, you can use `async`/`await` on a Firebase query (like `.get()`), which could clean this up instead of relying on callbacks.

Reply:
Do you know if it would technically be more performant to use a limit(1) to restrict the number of documents returned instead of adding a 2nd conditional? This is probably a useless optimization in this case and I prefer your suggestion because it seems functionally more valid, but I'm trying to understand the indexing behavior of Firestore. Adding a 2nd conditional in the query seems like it would still likely cause the query to have to potentially read through more of the index than what is necessary. If we really just cared about the existence of a single document owned by the user, limit() would ensure that we not only return a single document, but that we only have to read a single time from the field index correct?

Reply:
I believe that since Firestore is a NoSQL document store, it's actually more performant to use the `.where()`. I'm not positive about that, but I believe Firestore always has to check every document. It only applies the limit after having done so.

In terms of cost, however, Firestore only charges for the documents that are read or retrieved, so as long we know there's only ever going to be one item returned it doesn't matter whether we set a limit or not.